Feature reports stats

peter-budo opened the pull request:
- Battle reports in new processing, new design
- Some of previously reported issues from checkstyle, findbugs, pmd fixed
- Updated Android Gradle plugin
- Fixed stats tabs loading

peter-budo committed:
Added report stats json for testing

peter-budo committed:
Provided methods to build request URLs

peter-budo committed:
Parsing JSON

- Wrote class for game reports stats parsing
- Wrote simple JUnit test

peter-budo committed:
Fixed checkstyle issues as reported on build process

peter-budo committed:
Battle report UI changes

- Moved BattleReport adapter, fragment and listing fragment to stats package
- Removed BattleReport menu option from main menu
- Updated SoldierStatisticsActivity with BattleReportListingFragment
- Removed duplicated method from UrlFactory

peter-budo committed:
Updated Android Gradle plugin from 0.7.1 to 0.7.3

peter-budo committed:
New json for testing that has draw in results

peter-budo committed:
Battle reports initial layout change

peter-budo committed:
Merge branch 'dev' into feature-reports-stats

peter-budo committed:
Fixed bad resource names

peter-budo committed:
Merge branch 'dev' into feature-reports-stats

peter-budo committed:
UI changes of battle reports and data handling

- Moved from original SummaryBattleReport to GameReport
- Build list for display is using header and item objects
- Removed unused context parameter from DateTimeUtils

peter-budo committed:
Replaced loader manager call through activity with direct loader manager call

peter-budo committed:
Fixed weapon/vehicle/battlereports loading

peter-budo commented:
Build http://ci.peterscorner.co.uk/job/BF4-Intel-PR/13/

karllindmark commented:
@peter-budo: Can't access Jenkins, ha. :night_with_stars:

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/reports/BattleReportItem.java:
```diff
@@ -0,0 +1,81 @@
+                colour = R.color.battlereport_draw;
+                break;
+            default:
+                return R.color.black;
```
Any reason for this not assigning `colour` like the rest of the cases, instead of returning?

peter-budo replied:
I do not understand your question. I had to do this `context.getResources().getColor(colour);` otherwise colours were not displaying

karllindmark replied:
Yeah I know. But, on line 77, you're just returning `R.color.black` instead of assigning it to `colour`. :)

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/reports/BattleReportListingFragment.java:
```diff
@@ -126,14 +123,23 @@ private void setupListView(final View view) {
-        BattleReportAdapter reportAdapter = (BattleReportAdapter) getListAdapter();
-        if (reportAdapter == null) {
-            reportAdapter = new BattleReportAdapter(getActivity(), personaId);
+    private void sendDataToListView(final BattleReportStatistics statistics) {
```
I like the solution behind the headers&items. :+1:

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/utils/DateTimeUtils.java:
```diff
@@ -39,8 +38,7 @@ public static String toLiteral(final int seconds) {
     }
 
-    public static String toRelative(final Context context, final long timeInSeconds) {
+    public static String toRelative(final long timeInSeconds) {
```
I thought the `Context` parameter was used for localization, for some weird reason. I guess that's wrong? :cake:

peter-budo replied:
Check the code but I see no use of context either in line of code that is used neither of the commented bit that was removed

karllindmark replied:
Yeah, seems to work fine without it (which makes it easier for us).

karllindmark commented on BF4Intel/src/main/res/layout/list_item_battlereport.xml:
```diff
@@ -1,157 +1,79 @@
+      android:textSize="@dimen/text_xxl"
+      android:textStyle="bold"
+      android:layout_marginLeft="@dimen/spacing_l"
+      android:layout_marginRight="@dimen/spacing_l" />
```
In my opinion, the right margin should be removed for consistent look in cards (they have a basic padding).

peter-budo replied:
I will check what can be done with this

karllindmark replied:
I played around with the XML a bit when testing, and removing the right margin on this element looked a bit better, as the "cards" already have padding so to speak. Just my two :moneybag:

karllindmark commented on BF4Intel/src/main/res/layout/list_item_battlereport.xml:
```diff
@@ -1,157 +1,79 @@
+      android:layout_gravity="center_vertical">
+
+      <TextView
+        android:id="@+id/date"
```
I think it's better if we put the date below the round time, as it's not any of the "main data" of the card. Might need marginTop if we do so, not sure.

peter-budo replied:
I will look into it

karllindmark commented on core/src/main/java/com/ninetwozero/bf4intel/json/stats/reports/FavouriteGameReport.java:
```diff
@@ -0,0 +1,46 @@
+    @SerializedName("teams")
+    private Map<Integer, Players> teams = new HashMap<Integer, Players>();
+
+
```
A few too many white lines here I think.

peter-budo replied:
Will remove

karllindmark commented on core/src/main/java/com/ninetwozero/bf4intel/json/stats/reports/GameReport.java:
```diff
@@ -0,0 +1,53 @@
+        return serverName;
+    }
+
+    public abstract MatchResult matchResult(int soldierId);
```
`matchResult(int)`? What does it actually do? 

:-) :8ball:

peter-budo replied:
Will fix

karllindmark commented:
OK, general (non-related to a specific line of code):
- [ ] Only two out of three fragments display data at the same time. Usually either vehicles or reports that are at fault.
- [x] ...I didn't really have any others. :-)

peter-budo commented:
Test Jenkins I did some changes

peter-budo commented:
@karllindmark here is observation
1 if you click on stats, let weapon load completely and then wait few seconds the other 2 tabs will load fine
2 if you click on stats and once tabs displayed click on vehicles or reports that tab will fail to load. Also for the reason of clicking on it, the tab will get result from other loader. Therefore Gson will create empty object

This brings me to conclusion that action of clicking on tab and making it "visible" resets/cancels loader that was already fired. I will look into it tomorrow. We need to solve it as there will be one more tab "Details" and asking people to wait for loaders to finish is not a option

peter-budo committed:
Fixed tabs loading

- ViewPager added off screen page limit
- Removed application context by activity context

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/reports/BattleReportListingFragment.java:
```diff
@@ -126,14 +123,23 @@ private void setupListView(final View view) {
-        setListAdapter(reportAdapter);
+        itemList.add(new BattleReportHeader(R.string.header_latest_battlereport));
+        itemList.addAll(buildBaseItemList(new ArrayList<GameReport>(statistics.getStatsGameReports()), statistics.getSoldierId()));
+        BattleReportAdapter adapter = new BattleReportAdapter(getActivity().getApplicationContext(), itemList);
```
Side note: You shouldn't be using the application context here at moment - it messes up the styles right (ie, progressbar not drawing background for "non-achieved" part). 

@peter-budo

karllindmark replied:
(I guess the same goes for the other fragments as well)

peter-budo committed:
AndroidManifest changes

- Locked soldier stats in portrait
- Removed sdk declaration which is not used since it will be over writen by Gradle declarations

peter-budo committed:
Removed application context calls

peter-budo committed:
Base classes changes

 - Reversed BaseLoadingListFragment to previous state

peter-budo committed:
Off screen page limit set to 1

peter-budo committed:
Populate colour variable with black instead of returning directly

peter-budo committed:
Battlereport UI changes

- Removed match result padding
- Moved match date below match duration
- Changes styles between match duration and date

peter-budo committed:
Removed empty lines

peter-budo committed:
Method matchResult renamed to findMatchResultFor

peter-budo commented:
@karllindmark fixes as per request

karllindmark commented:
WHile I don't like the thought of the whole `setOffscreenLimit` solution (lower the value), it solves our issue at the current time being BUT will have to be looked at later on. Right now we focus on getting some other functionality ready.

karllindmark commented:
(Lowering the offscreen limit also makes an impact on unlocks)
